src/daemon.py

```diff
@@ -4,7 +4,7 @@
 
 import os
 import time
-from typing import List, Tuple
+from typing import List, Optional, Tuple
```

Comment:
This repo is using python 3.12. Even for some older version, the convention for type annotations has changed:
- `List`, `Tuple` should be imported. Instead, the build in types `list` and `tuple` should be used for type annotations.
- Instead of `Optional[T]` is is now the convention to use `T | None`. The import of `Optional` can be removed as well.